lib/rules/require-valid-css-selector-in-test-helpers.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const { getSourceModuleNameForIdentifier } = require('../utils/import');
+const { getPropertyValue, parseCallee, getAncestor } = require('../utils/utils');
+const { isImportDeclaration } = require('../utils/types');
+const cssTree = require('css-tree');
+
+const TEST_HELPER_IMPORTS = new Set(['fillIn', 'findAll', 'find', 'click', 'focus']);
+const QUERY_SELECTOR_METHODS = new Set(['querySelectorAll', 'querySelector']);
+const PARENT_NODE_NAMES = new Set(['element', 'document']);
+const SELECTOR_RULES = Object.freeze({
+  unclosedAttr: {
+    hasError: (selector) => selector.includes('[') && !selector.includes(']'),
```

Comment:
I'm unable to comment on the previous thread about this line, but I'd like to reply that we should look for errors even if they will be caught by tests. One of the main benefits of linting is that we get immediate feedback instead of having to wait for tests to run. Validating brackets seems like one of the basic checks we should perform, and we should not limit this check to looking for a single variation of mismatched brackets.

Reply:
Your previous example will be caught by this lint rule it just won't have the same error message or `fix` option.

Reply:
I've added a test case for `data-test]` to demonstrate the catching of other invalid attr cases.